Assets/Scripts/PointerToStar.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.Experimental.PlayerLoop;
+using UnityEngine.UI;
+using UnityEngine;
+using System.Collections;
+
+/// <summary>
+/// attach to an arrow icon image object under movableCanvas
+/// 
+/// 1. determine which stars to point to
+/// 2. have each star owning the canvas and the pointer image
+/// 3. make the image follow the star locations (lookat)
+/// 4. make those image the children of the canvas
+/// 5. make the cursor stay within the field of view
+///
+/// 6. when selecting the cursor, show a live image of the star
+///  within what diameter of distance?
+/// need to build a tree and then breath first search the closest stars ????
+/// simply, the 2 most adjacent stars
+/// </summary>
+public class PointerToStar : MonoBehaviour
+{
+    public GameObject StarToPoint; 
+    private bool _OnIsometricStarView;
+    public Camera cam;
+    public float imageWidth; 
+
+    public float disToClipPanel;
+    private bool _finishCameraPan;
+
+    private void Start()
+    {
+
+        // TODO: need to listen to the event of isometric cam. 
+        IsometricStarView.OnIsometricStarView += OnIsometricStarView;
+        GetComponent<Image>().enabled = false;
+        _finishCameraPan = false;
+        imageWidth = (float) (GetComponent<RectTransform>().rect.width * 0.5);
+    }
+
+
+    private void OnIsometricStarView(bool OnIsometricStarView)
+    {
+        _OnIsometricStarView = OnIsometricStarView;
+        print("star to point: " + StarToPoint.name);
+        // enable the image after the camera finish pan
+        Invoke(nameof(FinishPan), 4.0f);
+
+
+    }
+    
+    private void FinishPan()
+    {
+        _finishCameraPan = true;
+        GetComponent<Image>().enabled = true;
+
+    }
+
```

Comment:
Long method.

Create sub-methods, delegate tasks. Code isn't readable.

Reply:
The code is obsolete and will be deleted